### DISCUSSION TITLE
Weighted Voronoï diagram

### DISCUSSION THREAD

**Discussion by BastienGauthier** [shapely/shapely#2041] [General]
Hello !

Only a question for now, but if I wanted to add a weighted Voronoï diagram algorithm, what would be the right move ? Should I try to implement it in shapely, or directly in GEOS ?

Cheers,
Bastien

**Comment by jorisvandenbossche** [comment]
Shapely is mostly a wrapper around GEOS, and the actual geospatial algorithms live upstream in GEOS. So for something like, this would ideally be contributed to GEOS.

**Comment by BastienGauthier** [comment]
To complete the discussion, a workaround could be doing a n+1 dimension voronoi for a n dimension set of points.

See answers here : https://stackoverflow.com/questions/16024428/reference-algorithm-for-weighted-voronoi-diagrams
